Regression Mysql 8 support as the upgrade script do not use temporary table

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

storage

### Describe the bug

The fix of the issue https://github.com/keycloak/keycloak/issues/30631 in KC 26.0.7 broke the KC deployment on Mysql 8.0 with replication enabled.

Deploying KC 26.0.7 on a Mysql 8 replicated cluster fails with the error : 

```
     Reason: liquibase.exception.DatabaseException: The table does not comply with the requirements by an external plugin. [Failed SQL: (3098) DELETE FROM `adminop-db-rec`.USER_CONSENT WHERE ID IN (SELECT ID FROM TEMP_USER_CONSENT_IDS)]
```

As the table TEMP_USER_CONSENT_IDS is no more temporary, the above command failed.

We had to revert to KC 26.0.6


### Version

26.0.7

### Regression

- [X] The issue is a regression

### Expected behavior

We can upgrade or install KC 26 with mysql 8

### Actual behavior

Upgrade to KC 26.0.7 is broken. 

### How to Reproduce?

With a KC 24 deployement using a Mysql 8 cluster with replicaton enabled
Upgrade to KC 26.0.7

On startup the installation fails and a TEMP_USER_CONSENT_IDS persists in the database

### Anything else?

_No response_

actually in 26.0.7 I'm seeing:
```
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: io.quarkus.runtime.configuration.ConfigurationException: Persistence unit 'keycloak-default' was configured to run with a database version of at least '8.0.0', but the actual version is '5.7.0'. Consider upgrading your database. Alternatively, rebuild your application with 'quarkus.datasource.db-version=5.7.0' (but this may disable some features and/or impact performance negatively).
	at io.quarkus.hibernate.orm.runtime.service.QuarkusRuntimeInitDialectFactory.checkActualDbVersion(QuarkusRuntimeInitDialectFactory.java:70)
	at io.quarkus.hibernate.orm.runtime.observers.QuarkusSessionFactoryObserverForDbVersionCheck.sessionFactoryCreated(QuarkusSessionFactoryObserverForDbVersionCheck.java:15)
	at org.hibernate.internal.SessionFactoryObserverChain.sessionFactoryCreated(SessionFactoryObserverChain.java:35)
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:323)
	at io.quar
```
	
So I can't retest the 5.7 scenario on latest version. It also makes me think that since I'm no longer seeing the issue with TEMPORARY in 8.x in GCP then perhaps the best course is to reverse my first PR.

On a related note I think the release notes (https://www.keycloak.org/docs/latest/release_notes/index.html) should state mysql5.7 is no longer supported, I know there is the db versions docs but this is for tested versions not necessarily what is/is not supported, and until 25 mysql 5.7 was in fact working.

--- a/model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate25_0_0_MySQL_ConsentConstraints.java
+++ b/model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate25_0_0_MySQL_ConsentConstraints.java
@@ -42,6 +42,9 @@ public class JpaUpdate25_0_0_MySQL_ConsentConstraints extends CustomKeycloakTask
                         " max_dates ON uc.CLIENT_ID = max_dates.CLIENT_ID" +
                         " AND uc.USER_ID = max_dates.USER_ID AND uc.LAST_UPDATED_DATE = max_dates.MAX_UPDATED_DATE"
         ));
+        statements.add(new RawSqlStatement(
+                "ALTER TABLE TEMP_USER_CONSENT_IDS ADD PRIMARY KEY (ID)"
+        ));
         statements.add(new DeleteStatement(null, null, database.correctObjectName("USER_CONSENT", Table.class))
                 .setWhere("ID IN (SELECT ID FROM TEMP_USER_CONSENT_IDS)"));
         statements.add(new RawSqlStatement("DROP TABLE IF EXISTS TEMP_USER_CONSENT_IDS"));
@@ -61,6 +64,9 @@ public class JpaUpdate25_0_0_MySQL_ConsentConstraints extends CustomKeycloakTask
                         " FROM "+userConsentTable+" GROUP BY CLIENT_STORAGE_PROVIDER, EXTERNAL_CLIENT_ID, USER_ID HAVING COUNT(*) > 1 )" +
                         " max_dates ON uc.CLIENT_STORAGE_PROVIDER = max_dates.CLIENT_STORAGE_PROVIDER" +
                         " AND uc.EXTERNAL_CLIENT_ID = max_dates.EXTERNAL_CLIENT_ID AND uc.USER_ID = max_dates.USER_ID AND uc.LAST_UPDATED_DATE = max_dates.MAX_UPDATED_DATE;"
+        ));
+        statements.add(new RawSqlStatement(
+                "ALTER TABLE TEMP_USER_CONSENT_IDS2 ADD PRIMARY KEY (ID)"
         ));
         statements.add(new DeleteStatement(null, null, database.correctObjectName("USER_CONSENT", Table.class))
                 .setWhere("ID IN (SELECT ID FROM TEMP_USER_CONSENT_IDS2)"));

